components/script/dom/element.rs

```diff
@@ -213,10 +212,8 @@ pub enum AdjacentPosition {
     BeforeEnd,
 }
 
-impl<'a> TryFrom<&'a str> for AdjacentPosition {
-    type Error = Error;
-
-    fn try_from(position: &'a str) -> Result<AdjacentPosition, Self::Error> {
+impl AdjacentPosition {
+    fn try_from(position: &str) -> Result<AdjacentPosition, Error> {
```

Comment:
Could you use [`FromStr`](https://doc.rust-lang.org/std/str/trait.FromStr.html) here?